sunpy/time/time.py

```diff
@@ -217,54 +218,40 @@ def convert_time_astropy(time_string, **kwargs):
     return time_string
 
 
-@convert_time.register(list)
-def convert_time_list(time_list, format=None, **kwargs):
-    item = time_list[0]
-    # If we have a list of strings, need to get the correct format from our
-    # list of custom formats.
-    if isinstance(item, str) and format is None:
-        string_format = _get_time_fmt(item)
-        return Time.strptime(time_list, string_format, **kwargs)
-
-    # Otherwise return the default method
-    return convert_time.dispatch(object)(time_list, format, **kwargs)
-
-
 @convert_time.register(str)
+@convert_time.register(list)
 def convert_time_str(time_string, **kwargs):
-    if 'TAI' in time_string:
-        kwargs['scale'] = 'tai'
+    is_single_string = isinstance(time_string, str)
+    is_string_list = isinstance(time_string, list) and isinstance(time_string[0], str)
 
-    for time_format in TIME_FORMAT_LIST:
-        try:
+    if is_single_string or is_string_list:
+        first_item = time_string[0] if is_string_list else time_string
+        if 'TAI' in first_item:
+            kwargs['scale'] = 'tai'
+
+        for time_format in TIME_FORMAT_LIST:
             try:
-                ts, add_one_day = _regex_parse_time(time_string, time_format)
-            except TypeError:
-                break
-            if ts is None:
-                continue
-            t = Time.strptime(ts, time_format, **kwargs)
-            if add_one_day:
-                t += _ONE_DAY_TIMEDELTA
-            return t
-        except ValueError:
-            pass
+                try:
+                    ts, add_one_day = _regex_parse_time(first_item, time_format)
+                except TypeError:
+                    break
+                if ts is None:
+                    continue
+                if is_single_string:
+                    t = Time.strptime(ts, time_format, **kwargs)
+                    if add_one_day:
+                        t += _ONE_DAY_TIMEDELTA
+                else:
+                    # For a list of strings, we do not try to correct 24:00:00
```

Comment:
Why?

Reply:
When parsing a single string, we check for "24:00:00" at the same time we check if a string format matches because we need to modify the string before passing it to `strptime()`.  However, for a list of strings, which are typically all of the same string format, we currently take the performance shortcut of looking at only the first element to determine which string formats to try.  That means we never actually look at other elements.

Expanding support to a list is certainly doable, but the performance impact of looking at every element in a long list for "24:00:00" should not be accepted lightly, given how uncommon it should be.  Maybe a keyword option?  (The implementation would be quite similar to also supporting mixed-format lists.)